railties/lib/rails/generators/rails/resource/resource_generator.rb

```diff
@@ -19,7 +19,7 @@ class ResourceGenerator < ModelGenerator # :nodoc:
 
       class << self
         def desc(description = nil)
-          ERB.new(File.read(usage_path)).result(binding)
+          @desc ||= ERB.new(File.read(usage_path)).result(binding)
```

Comment:
While this change doesn't directly tackle the primary concern of this PR, it introduces caching to accommodate Thor's pattern of invoking desc twice, aligning with the caching mechanism found in Rails::Generators::Base.

https://github.com/rails/thor/blob/a43d92fad7ebd77d359b7b96eb3db8a73ef9057c/lib/thor/group.rb#L34
https://github.com/rails/rails/blob/f7353283fd771e99b9675bcb9394e398964195f8/railties/lib/rails/generators/base.rb#L44-L45